org.adempiere.base/src/org/compiere/acct/Doc_InOut.java

```diff
@@ -199,6 +199,11 @@ public ArrayList<Fact> createFacts (MAcctSchema as)
 				Map<String, BigDecimal> batchLotCostMap = null;
 				DocLine_InOut line = (DocLine_InOut) p_lines[i];				
 				MProduct product = line.getProduct();
+				//If expense type stocked product, no impact on COGS as it not deducting inventory

```

Comment:
when doing a manual material receipt - is complaining with "No Costs for ..." - if the expense product is not to be posted, then the posting must not throw error

Reply:
We were creating cost. now creating new PR which has no cost created at all and so it will not throw any error.